lib/broadside/deploy/ecs_deploy.rb

```diff
@@ -202,85 +202,78 @@ def create_task_definition(name, options = {})
 
     # reloads the service using the latest task definition
     def update_service
-      td = get_latest_task_def_id
-      debug "Updating #{family} with scale=#{@deploy_config.scale} using task #{td}..."
-      resp = ecs_client.update_service({
+      task_definition_arn = get_latest_task_definition_arn
+      debug "Updating #{family} with scale=#{@deploy_config.scale} using task #{task_definition_arn}..."
+
+      update_service_response = ecs_client.update_service(
         cluster: config.ecs.cluster,
         service: family,
-        task_definition: get_latest_task_def_id,
+        task_definition: task_definition_arn,
         desired_count: @deploy_config.scale
-      })
+      )
 
-      if resp.successful?
-        begin
-          ecs_client.wait_until(:services_stable, {cluster: config.ecs.cluster, services: [family]}) do |w|
-            w.max_attempts = @deploy_config.timeout.nil? ? @deploy_config.timeout : @deploy_config.timeout / config.ecs.poll_frequency
-            w.delay = config.ecs.poll_frequency
-            seen_event = nil
-            w.before_wait do |attempt, response|
-              debug "(#{attempt}/#{w.max_attempts}) Polling ECS for events..."
-              # skip first event since it doesn't apply to current request
-              if response.services[0].events.first &&
-                  response.services[0].events.first.id != seen_event &&
-                  attempt > 1
-                seen_event = response.services[0].events.first.id
-                debug(response.services[0].events.first.message)
-              end
-            end
+      exception 'Failed to update service during deploy.' unless update_service_response.successful?
+
+      ecs_client.wait_until(:services_stable, { cluster: config.ecs.cluster, services: [family] }) do |w|
+        w.max_attempts = @deploy_config.timeout ? @deploy_config.timeout / config.ecs.poll_frequency : nil
+        w.delay = config.ecs.poll_frequency
+        seen_event = nil
+
+        w.before_wait do |attempt, response|
+          debug "(#{attempt}/#{w.max_attempts}) Polling ECS for events..."
+          # skip first event since it doesn't apply to current request
+          if response.services[0].events.first && response.services[0].events.first.id != seen_event && attempt > 1
+            seen_event = response.services[0].events.first.id
+            debug(response.services[0].events.first.message)
           end
-        rescue Aws::Waiters::Errors::TooManyAttemptsError
```

Comment:
i was thinking the exception would bubble up and create an exception anyways, just of the type `Aws::Waiters::Errors::TooManyAttemptsError` - which is actually more accurate - than the `Broadside::Error` or whatever.

in general i'm not a fan of squashing a bunch of useful information about exceptions by removing their class and stuffing them in a single exception class; it's caused me some pain debugging.